server/sonar-server/src/test/java/org/sonar/server/qualitygate/ws/SearchActionTest.java

```diff
@@ -0,0 +1,180 @@
+        tuple("gateId", true),
+        tuple("query", false),
+        tuple("organization", false),
+        tuple("selected", false),
+        tuple("page", false),
+        tuple("pageSize", false));
+  }
+
+  @Test
+  public void search_projects_of_a_quality_gate_from_an_organization() {
+    OrganizationDto organization = db.organizations().insert();
+    ComponentDto project = db.components().insertPublicProject(organization);
+    QualityGateDto qualityGate = db.qualityGates().insertQualityGate(organization);
+    associateProjectToQualityGate(project, qualityGate);
+    userSession.addPermission(ADMINISTER_QUALITY_GATES, organization);
+
+    SearchResponse response = ws.newRequest()
+      .setParam(PARAM_GATE_ID, valueOf(qualityGate.getId()))
+      .setParam(PARAM_ORGANIZATION, valueOf(organization.getKey()))
+      .executeProtobuf(SearchResponse.class);
+
+    assertThat(response).isNotNull();
+    assertThat(response.getMore()).isFalse();
+    assertThat(response.getResultsCount()).isEqualTo(1);
+    assertThat(response.getResults(0).getId()).isEqualTo(project.getId());
+    assertThat(response.getResults(0).getName().substring(4)).isEqualTo(project.getKey().substring(3));
+  }
+
+  @Test
+  public void search_on_default_organization_when_none_is_provided(){
+    OrganizationDto defaultOrganization = db.getDefaultOrganization();
+
+    ComponentDto project = db.components().insertPublicProject(defaultOrganization);
+    QualityGateDto qualityGate = db.qualityGates().insertQualityGate(defaultOrganization);
+    associateProjectToQualityGate(project, qualityGate);
+    userSession.addPermission(ADMINISTER_QUALITY_GATES, defaultOrganization);
+
+    SearchResponse response = ws.newRequest()
+      .setParam(PARAM_GATE_ID, valueOf(qualityGate.getId()))
+      .executeProtobuf(SearchResponse.class);
+
+    assertThat(response).isNotNull();
+    assertThat(response.getMore()).isFalse();
+    assertThat(response.getResultsCount()).isEqualTo(1);
+    assertThat(response.getResults(0).getId()).isEqualTo(project.getId());
+    assertThat(response.getResults(0).getName().substring(4)).isEqualTo(project.getKey().substring(3));
+  }
+
+  @Test
+  public void fail_when_quality_gates_does_not_belong_to_organization(){
+    OrganizationDto organization = db.organizations().insert();
+    OrganizationDto otherOrganization = db.organizations().insert();
+    QualityGateDto qualityGate = db.qualityGates().insertQualityGate(otherOrganization);
+
+    expectedException.expect(NotFoundException.class);
+    expectedException.expectMessage(format("Quality gate '%s' does not exists.", qualityGate.getId()));
+
+    SearchResponse response = ws.newRequest()
+      .setParam(PARAM_GATE_ID, valueOf(qualityGate.getId()))
+      .setParam(PARAM_ORGANIZATION, valueOf(organization.getKey()))
+      .executeProtobuf(SearchResponse.class);
+
+  }
+
+
+  @Test
+  public void return_parameter_more_is_true_when_not_all_project_fit_in_page_size() {
```

Comment:
name is not clear, something like this would be better "more_is_true_when_not_all_project_fit_in_page_size", as more is not the parameter but it's a field in the response